randovania/game_description/resources/resource_database.py

```diff
@@ -33,6 +33,12 @@ def default_base_damage_reduction(db: ResourceDatabase, current_resources: Resou
 )
 
 
+@dataclasses.dataclass(frozen=True)
+class NamedRequirementTemplate:
+    requirement: Requirement
+    display_name: str | None
```

Comment:
I don't think i quite get this. The display name is more or less a "long key". So why make this "str | None" and deal with all the checking if its not null, instead of just forcing it to be a string?

Reply:
My thought was that given a lot of existing templates use the key as a display name, no need to duplicate things. But maybe that's just better

Reply:
It does make the code easier to understand IMO.
Worst case, if someone really doesnt like their new template shortkey, they can just always write a migration to get rid of it :P